datareservoirio/_logging.py

```diff
@@ -41,3 +41,25 @@ def wrapper(self, *args, **kwargs):
             raise e
 
     return wrapper
+
+
+def log_retry(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except Exception as e:
+            properties = {
+                "customDimensions": {
+                    "drioPackage": f"python-datareservoirio/{drio.__version__}",
+                    "message": "exception is retried",
+                }
+            }
+            if os.getenv(ENV_VAR_ENGINE_ROOM_APP_ID) is not None:
+                properties["customDimensions"]["engineRoomAppId"] = os.getenv(
+                    ENV_VAR_ENGINE_ROOM_APP_ID
+                )
+            exceptions_logger.warning(e, extra=properties)
```

Comment:
Since the only difference here really is the warning vs error (well, and the exceptions is retried message), as a decorator it could be written as one function that takes this optional message and the log-level as input, but they way it seems to be done in Python is to wrap the entire thing in another function. As long as there are only to variants like here, and likely other logging will be used with the logger directly and not through more attributes, I prefer to do it the way it is done here, but I just mention it as an optional (but in my opinion not preferrable) way do implement it https://stackoverflow.com/questions/10176226/how-do-i-pass-extra-arguments-to-a-python-decorator